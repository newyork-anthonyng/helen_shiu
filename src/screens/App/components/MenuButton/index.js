import React, { Children, cloneElement } from 'react';
import T from 'prop-types';
import noop from '../../utilities/noop';

const MenuButton = ({ children, onClick, onKeyDown }) => {
  return cloneElement(Children.only(children), {
    onClick,
    onKeyDown,
    tabIndex: 0,
  });
};

MenuButton.defaultProps = {
  onClick: noop,
  onKeyDown: noop,
};

MenuButton.propTypes = {
  onClick: T.func.isRequired,
  onKeyDown: T.func.isRequired,
};

export default MenuButton;
