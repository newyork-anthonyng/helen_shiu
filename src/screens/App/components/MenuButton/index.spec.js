import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MenuButton from './';
import noop from '../../utilities/noop';

const defaultProps = {
  onClick: noop,
  onKeyDown: noop,
};

it('should render correctly', () => {
  const wrapper = shallow(
    <MenuButton {...defaultProps}>
      <h1>Open Menu</h1>
    </MenuButton>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should run callback when clicking on button', () => {
  const cb = jest.fn();
  const wrapper = shallow(
    <MenuButton {...defaultProps} onClick={cb}>
      <h1>Open Menu</h1>
    </MenuButton>
  );

  wrapper.simulate('click');

  expect(cb).toHaveBeenCalledTimes(1);
});

it('should run callback when pressing key', () => {
  const cb = jest.fn();
  const wrapper = shallow(
    <MenuButton {...defaultProps} onKeyDown={cb}>
      <h1>Open Menu</h1>
    </MenuButton>
  );

  wrapper.simulate('keydown');

  expect(cb).toHaveBeenCalledTimes(1);
});
