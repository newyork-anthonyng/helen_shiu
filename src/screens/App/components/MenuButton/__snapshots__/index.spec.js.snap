// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<h1
  onClick={[Function]}
  onKeyDown={[Function]}
  tabIndex={0}
>
  Open Menu
</h1>
`;
